src/components/GooglePayButton.tsx

```diff
@@ -0,0 +1,84 @@
+import { type StyleProp, type ViewStyle } from 'react-native';
+import { useCallback } from 'react';
+import { PlatformPayView } from '..';
+
+export const GooglePayButtonTheme = {
+  DARK: 1,
+  LIGHT: 2,
+};
+
+export const GooglePayButtonType = {
+  BUY: 1,
+  BOOK: 2,
+  CHECKOUT: 3,
+  DONATE: 4,
+  ORDER: 5,
+  PAY: 6,
+  SUBSCRIBE: 7,
+  PLAIN: 8,
+};
+
+export interface GooglePayButtonProps {
+  theme?: keyof typeof GooglePayButtonTheme;
+  type?: keyof typeof GooglePayButtonType;
+  radius?: number;
+  onPress?: () => void;
+  style?: StyleProp<ViewStyle>;
+}
+
+function getButtonTheme(type: string): number {
+  switch (type) {
+    case 'DARK':
+      return 1;
+    case 'LIGHT':
+      return 2;
+    default:
+      return 1;
+  }
+}
+
+function getButtonType(type: string): number {
```

Comment:
```suggestion
function getButtonType(type: keyof typeof GooglePayButtonType): number {
```

Reply:
Addressed in [2045d8e](https://github.com/Adyen/adyen-react-native/pull/701/commits/2045d8e83db5674ef9260ef3965040e00162ddb1)